feeluown/containers/player_control_panel.py

```diff
@@ -27,9 +27,61 @@ class SongBriefLabel(QLabel):
 
     def __init__(self, app):
         super().__init__(text=self.default_text, parent=None)
+        self.setSizePolicy(QSizePolicy.Expanding, QSizePolicy.Preferred)
         self._app = app
+        self.t = QTimer()
+        self.txt = self.raw_text = '...'
+        self.font_metrics = QFontMetrics(QApplication.font())
+        self.text_rect = self.font_metrics.boundingRect(self.raw_text)
+        self.newX = 0
+
+        self.t.timeout.connect(self.change_text_position)
         self._fetching_artists = False
 
+    def change_text_position(self):
+        if not self.parent().isVisible():
+            self.t.stop()
+            self.newX = 0
+            return
+        if self.text_rect.width() + self.newX > 0:
+            # control the speed of rolling
+            self.newX -= 5
+        else:
+            self.newX = self.width()
+        self.update()
+
+    def setText(self, text):
+        self.txt = self.raw_text = text
+        self.text_rect = self.font_metrics.boundingRect(self.raw_text)
+        self.newX = 0
+        self.update()
+
+    def enterEvent(self, event):
+        if self.txt != self.raw_text:
```

Comment:
how about comparing the `text_rect.width` with `self.width`?

Reply:
> how about comparing the `text_rect.width` with `self.width`?

`QFontMetrics.boundingRect` is not so accurate. When the `text_rect`'s width is only a little smaller, you will get a "xxxx..." song title but cannot roll.

Reply:
Could you please add this comment as code comment. It will be a really helpful message for following developers.